File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
match = {}
for i in ["()", "{}", "[]"]:
    a,b = list(i)
    match[b] = a
    
def isCorrect(s):
    stk = []
    for i in s:
        if not stk or i in match.values():
            stk.append(i)
            continue
            
        if i in match.keys() and match[i] == stk[-1]:
            stk.pop()
        else:
            return False
        
    return len(stk) == 0
        
                

def solution(s):
    cnt = 0
    for i in range(len(s)):
        _s = s[i:] + s[:i]
        if isCorrect(list(_s)):
            cnt += 1
            
    return cnt

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
let map = {
    ["{"] : "}",
    ["["] : "]",
    ["("] : ")",
    ["}"] : "{",
    ["]"] : "[",
    [")"] : "("
}

function isCollect(s){
    let stk = [];
    let close = "}])";
    
    for(let v of s){
        if(close.includes(v) && map[v] === stk.at(-1)) {
            stk.pop();
            continue;
        }
        stk.push(v)
    }
    
    return stk.length === 0 ? true : false
}

function solution(s) {
    let cnt = 0;
    for(let i = 0; i < s.length; i++){
        cnt += isCollect(s) ? 1 : 0;
        s = s.slice(1) + s[0]
    }
    
    return cnt;
}